### PR TITLE
add helm chart

### PR DESCRIPTION
**Pull request by tobytoyuito**

### CODE REVIEW

**Comment by stshrive** [comment]
Can you merge with master now that my recent PR has gone through? This should fix the failing checks.

**Review by stshrive — APPROVED** [review]

File: helm-charts/vsts-agent/values.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Default values for vsts-agent.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 2
+
+image:
+  repository: airsharedcontainers.azurecr.io/kube-agent
+  # tag has to be a string, please quote it
+  tag: "1094096"
```
Comment (stshrive) [inline helm-charts/vsts-agent/values.yaml:10]: So this is templatizing the tag?

File: helm-charts/vsts-agent/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "vsts-agent.fullname" -}}
+  {{- if .Values.fullnameOverride -}}
+    {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+  {{- else -}}
+    {{- $name := default .Chart.Name .Values.nameOverride -}}
+    {{- if contains $name .Release.Name -}}
+      {{- .Release.Name | trunc 63 | trimSuffix "-" -}}
```
Comment (stshrive) [inline helm-charts/vsts-agent/templates/_helpers.tpl]: trunc 63 looks awfully suspicious. How do we find these things if they get truncated?

Comment (stshrive) [reply helm-charts/vsts-agent/templates/_helpers.tpl]: You pipe into trunc and trim a lot. Is there a better way to limit these strings?

File: helm-charts/vsts-agent/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A custom VSTS agent container with Python3, kubectl and msftkube preinstalled.
+name: vsts-agent
+version: 0.1.1094096
```
Comment (stshrive) [inline helm-charts/vsts-agent/Chart.yaml:5]: Version and tag (in values.yaml) look very related. Do they need to change in tandem in these files?

File: helm-charts/vsts-agent/values.yaml (hunk 1 above)
Comment (tobytoyuito) [reply helm-charts/vsts-agent/values.yaml:10]: yes. This is the default one. Values in this file can be customized during install

File: helm-charts/vsts-agent/Chart.yaml (hunk 3 above)
Comment (tobytoyuito) [reply helm-charts/vsts-agent/Chart.yaml:5]: yes. This is the chart version. It helps the consumers to know what is the default image tag if they don't specify one

File: helm-charts/vsts-agent/templates/_helpers.tpl (hunk 2 above)
Comment (tobytoyuito) [reply helm-charts/vsts-agent/templates/_helpers.tpl]: This file is auto generated by helm create. I think vsts agent doesn't have any DNS related stuff. I'll remove the trunc.

**Pull request merged**